File: pyjs/pyjamas-0.1/pyjs/tests/test012.py
import DOM

class Widget:

    def __init__(self):
        self.attached = False

    def setParent(self, parent):
        self.parent = parent
        if parent == None:
            self.onDetach()
        elif parent.attached:
            self.onAttach()

    def onAttach(self):
        if self.attached:
            return
        self.attached = True
        DOM.setEventListener(self.getElement(), self)
        
    def onDetach(self):
        if not self.attached:
            return
        self.attached = False
        DOM.setEventListener(self.getElement(), None)

File: pyjs/pyjamas-0.1/pyjs/tests/test012.js
function Widget() {
    return new __Widget();
}
function __Widget() {
    this.attached = false;
}
__Widget.prototype.setParent = function(parent) {
    this.parent = parent;
    if ((parent == null)) {
    this.onDetach();
    }
    else if (parent.attached) {
    this.onAttach();
    }
};
__Widget.prototype.onAttach = function() {
    if (this.attached) {
    return;
    }
    this.attached = true;
    DOM_setEventListener(this.getElement(), this);
};
__Widget.prototype.onDetach = function() {
    if (!(this.attached)) {
    return;
    }
    this.attached = false;
    DOM_setEventListener(this.getElement(), null);
};
